.changelog/unreleased/improvements/3211-make-cs-reactor-no-longer-takes-cs-locks.md

```diff
@@ -0,0 +1,4 @@
+- `[consensus]` Make the consensus reactor no longer have packets on receive take the consensus lock.
+Consensus will now update the reactor's view after every relevant change through the existing event
```

Comment:
This is not really an event bus.

I would say that we use the synchronous events produced by the state to the reactor to transmit a fresh/updated copy of the consensus state to the reactor.

Reply:
Ok! I thought this event system was called the event bus, thats my bad. Thank you!

Reply:
No problem, it is described like that:

https://github.com/cometbft/cometbft/blob/2dd5209177270fde125c71aeedda9819046f92f0/internal/consensus/state.go#L131-L133

Reply:
And the comment is outdated, as we have a third event, eheheheheh